addons/trenches/functions/fnc_initTrench.sqf

```diff
@@ -0,0 +1,31 @@
+/*
+ * Author: chris579
+ * Inits a trench
+ *
+ * Arguments:
+ * 0: trench <OBJECT>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [TrenchObj] call ace_trenches_fnc_initTrench
+ *
+ * Public: No
+ */
+#include "script_component.hpp"
+params [
+    ["_object", objnull, [objNull]]
+];
+
+if (is3DEN) exitWith {
+    [_object] call FUNC(initTrench3DEN);
+};
+
+if (local _object) then {
+    // Has to be spawned to ensure MP compatibility
```

Comment:
Can vouch for this. Same reason setVariable public fails from init. initPost is no option without creating even more event handlers, because 3den compat.